src/nvim/tui/tui.c

```diff
@@ -527,7 +529,13 @@ static void tui_flush(UI *ui)
   }
 
   unibi_goto(ui, data->row, data->col);
+
+  if (!data->busy) {
+    unibi_out(ui, unibi_cursor_normal);
+  }
+
   flush_buf(ui);
+  unibi_out(ui, unibi_cursor_invisible);
```

Comment:
> In the patch invis_buf is not flushed: 

Here's a correction to apply on top of the previous patch:

```diff
diff --git a/src/nvim/tui/tui.c b/src/nvim/tui/tui.c
index c25ec28..e320ef7 100644
--- a/src/nvim/tui/tui.c
+++ b/src/nvim/tui/tui.c
@@ -345,21 +345,18 @@ static void tui_cursor_goto(UI *ui, int row, int col)
   data->row = row;
   data->col = col;
   unibi_goto(ui, row, col);
-  data->require_invisible_cursor = true;
 }

 static void tui_busy_start(UI *ui)
 {
   TUIData *data = ui->data;
   data->busy = true;
-  data->require_invisible_cursor = true;
 }

 static void tui_busy_stop(UI *ui)
 {
   TUIData *data = ui->data;
   data->busy = false;
-  data->require_invisible_cursor = true;
 }

 static void tui_mouse_on(UI *ui)
@@ -483,7 +480,6 @@ static void tui_scroll(UI *ui, int count)
     invalidate(ui, data->scroll_region.top, data->scroll_region.bot,
         data->scroll_region.left, data->scroll_region.right);
   }
-  data->require_invisible_cursor = true;
 }

 static void tui_highlight_set(UI *ui, HlAttrs attrs)
@@ -642,6 +638,7 @@ static void unibi_goto(UI *ui, int row, int col)
   data->params[0].i = row;
   data->params[1].i = col;
   unibi_out(ui, unibi_cursor_address);
+  data->require_invisible_cursor = true;
 }

 static void unibi_out(UI *ui, int unibi_index)
@@ -794,25 +791,25 @@ static void flush_buf(UI *ui)
 {
   uv_write_t req;
   uv_buf_t buf[2];
-  int buf_index = 0;
+  unsigned int buf_count = 1;
   char invis_buf[64];
   TUIData *data = ui->data;

-  if (data->require_invisible_cursor) {
-    if (!data->busy) {
-      unibi_out(ui, unibi_cursor_normal);
-    }
-
+  if (data->require_invisible_cursor || data->busy) {
     const char *str = unibi_get_str(data->ut, unibi_cursor_invisible);
     buf[0].base = invis_buf;
     buf[0].len = unibi_run(str, data->params, invis_buf, sizeof(invis_buf));
-    buf_index++;
+    buf_count++;
     data->require_invisible_cursor = false;
+
+    if (!data->busy) {
+      unibi_out(ui, unibi_cursor_normal);
+    }
   }

-  buf[buf_index].base = data->buf;
-  buf[buf_index].len = data->bufpos;
-  uv_write(&req, (uv_stream_t *)&data->output_handle, buf + buf_index, 1, NULL);
+  buf[buf_count - 1].base = data->buf;
+  buf[buf_count - 1].len = data->bufpos;
+  uv_write(&req, (uv_stream_t *)&data->output_handle, buf, buf_count, NULL);
   uv_run(data->write_loop, UV_RUN_DEFAULT);
   data->bufpos = 0;
 }
```

Note that this variant only sets `require_invisible_cursor` when the cursor is moved with `unibi_goto`.

Your'e right though, your patch is simple and does not seem introduce unwanted cursor flashes. Can you make a PR with this patch?